psi4/src/psi4/lib3index/dfhelper.cc

```diff
@@ -1886,16 +1881,15 @@ void DFHelper::transform() {
 void DFHelper::first_transform_pQq(size_t nao, size_t naux, size_t bsize, size_t bcount, size_t block_size,
     double* Mp, double* Tp, double* Bp, std::vector<std::vector<double>>& C_buffers){
 
-    size_t rank = 0;
-
     // perform first contraction on pQq, thread over p.
-    #pragma omp parallel for firstprivate(nao, naux, bsize, block_size, rank) schedule(guided) num_threads(nthreads_)
-    for (size_t k = 0; k < nao_; k++) {
```

Comment:
The `firstprivate` here with `nao`, `naux`, `bsize` and `block_size` guaranteed all threads got their own copy. Not sure how much (if at all) this affects latency of memory fetches, but that's why it was used.   The cleanup of most other pragmas is definitely warranted.  Of course defining the variables within the loops makes them private. Just a note.

Reply:
Also worth mentioning this is one of the main computational kernels.

Reply:
Which tests could used to measure the impact to the performance?

Reply:
Also `nao` and `naux` are not even used in the OpenMP block!

Reply:
touché!  The tests would require timings of the different versions.. but it might be easiest just to keep it in.  Maybe someone (possibly at GT) can play with some of these ideas and get the best configuration.  This is the only `private` or `firstprivate` that caught my eye as being useful.